crates/sargon/src/profile/v100/factors/factor_source_id_from_hash.rs

```diff
@@ -89,6 +88,15 @@ impl FactorSourceIDFromHash {
             mnemonic_with_passphrase,
         )
     }
+
+    pub fn new_for_passphrase(
```

Comment:
this is not needed right, and in fact, it opens up for the possibility of discrepancies..? 

We should do:
```rust
DeviceFactorSource::sample().id
DeviceFactorSource::sample_other().id

LedgerFactorSource::sample().id
LedgerFactorSource::sample_other().id


ArculusFactorSource::sample().id
ArculusFactorSource::sample_other().id
```

etc below